### ISSUE TITLE
Zoom: center on cursor

### ISSUE THREAD

**Issue by tobiasBora** [xournalpp/xournalpp#2054]
**Is your feature request related to a problem? Please describe.**

Usually, when I want to zoom to a specific part of the image, I just put my mouse pointer there, and zoom with `Ctrl + scrool`. In most programs (see Gimp for example, or google maps...), it does a zoom "centered" on the mouse pointer (i.e. the screen is zoomed/scrolled such that the point under the pointer is always the same on screen). However in xournalapp the zoom is centered in the middle of the screen, so zooming to a specific point requires 2 steps : zoom + scrool instead of simply zooming, which is a bit annoying. 

**Describe the solution you'd like**

Just scroll automatically the picture after a zoom to make sure the point under the mouse pointer does not change.

Thanks!

**Comment by Febbe** [comment]
The thing is, the document is always centred, so it will perform a centred zoom until the document is larger than the view. 
Currently the zooming is a bunch of plasters holding everything together. It will require a complete rewrite.

**Comment by TentativeConvert** [comment]
I'm on Ubuntu 20.04, using the nightly builds of Xournal++ available via the PPA, updated today (6 August). The About page of Xournal++ reads: `Version 1.1.0+dev`, `nogitfound`, `Built on Aug 5 2020, 20:45:45` `with libgtk 3.24.20`.  

Zooming works as expected when I use the slider in the toolbar.  However, zooming in and out *with the keyboard* (Ctrl +/-) yields unpredicable jumping behaviour.  I start zooming in on, say, page 42 of a document, and end up on page 8.  I zoom out again and I end up on page 17.

**Comment by Febbe** [comment]
@rolandlo, did you merged a pull request, regarding scrolling the last 3 mounts?

**Comment by rolandlo** [comment]
@Febbe No, I did not. The pull requests I merged were all related to pdf export, mouse cursor, page count, audio sampling rate, object selection and toolbar hiding. 
I think the last commit related to scrolling is #2023 from June 10 (merged by you).

**Comment by rolandlo** [comment]
In fact I cannot see any difference in the zooming behavior from #2023 and the current version. I can confirm that the zooming behavior is strange in that you may start zooming in on page 42 and end up on page 8. For me that happens regardless of whether I use the zooming via the slider or via keyboard.

**Comment by TentativeConvert** [comment]
I confirm (correcting my previous post) that zooming with the slider gives the same jumping behaviour, at least some of the time.

**Comment by arjunkc** [comment]
Thought I'd chime in and mention that I have this zooming issue as well. When you zoom into a page, you jump to another page (which doesn't happen in xournal).

**Comment by Greenscreener** [comment]
I am also experiencing this issue on version 1.0.20 on openSUSE Tumbleweed.

**Comment by frogtd** [comment]
Ctrl + +/- works for me but the ctrl + scroll also scrolls the page while zooming on arch

**Comment by personalizedrefrigerator** [comment]
> Ctrl + +/- works for me but the ctrl + scroll also scrolls the page while zooming on arch

Is this with `1.1.0+dev`?

( I would also like to link #2837 to this; it changes ctrl+scroll zoom behavior )
